Generals/Code/GameEngine/Source/Common/System/SaveGame/GameState.cpp

```diff
@@ -933,6 +933,13 @@ AsciiString GameState::portableMapPathToRealMapPath(const AsciiString& in) const
 		return AsciiString::TheEmptyString;
 	}
 
+	// TheSuperHackers @security bobtista 06/11/2025 Validate file extension to prevent arbitrary file types
+	if (!FileSystem::hasValidTransferFileExtension(prefix))
```

Comment:
Conceptually this test does not belong into this function. It probably needs to be moved into `ConnectionManager::processFile`.